server/index.js

```diff
@@ -33,6 +33,25 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const productId = req.params.productId;
+  const sql = `
+    select *
+      from "products"
+    where "productId" = $1
+  `;
+  const values = [productId];
+
+  db.query(sql, values)
+    .then(result => {
+      if (!result.rows.productId) {
+        next(new ClientError(`cannot ${req.method} ${req.originalUrl}`, 404));
+      }
```

Comment:
`result.rows` is an array and arrays do not have a `productId` property. You want to check if the `result.rows` is empty or not.
Also, be sure to include a `return` statement so that your server doesn't try to respond twice and crash.